x/fees/ante/ante.go

```diff
@@ -0,0 +1,80 @@
+package ante
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	cosmosante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	"github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/desmos-labs/desmos/x/fees"
+)
+
+// NewAnteHandler returns a custom AnteHandler that besides all the default checks
+//(sequence number increment, signature and account number checks, fee deduction) make sure that each
+// transaction has a minimum fee based on messages types
+func NewAnteHandler(
+	ak keeper.AccountKeeper,
+	supplyKeeper types.SupplyKeeper,
+	sigGasConsumer cosmosante.SignatureVerificationGasConsumer,
+	feesKeeper fees.Keeper,
+) sdk.AnteHandler {
+	return sdk.ChainAnteDecorators(
+		cosmosante.NewSetUpContextDecorator(), // outermost AnteDecorator. SetUpContext must be called first
+		cosmosante.NewMempoolFeeDecorator(),
+		cosmosante.NewValidateBasicDecorator(),
+		cosmosante.NewValidateMemoDecorator(ak),
+		NewMinFeeDecorator(feesKeeper),
+		cosmosante.NewConsumeGasForTxSizeDecorator(ak),
+		cosmosante.NewSetPubKeyDecorator(ak), // SetPubKeyDecorator must be called before all signature verification decorators
+		cosmosante.NewValidateSigCountDecorator(ak),
+		cosmosante.NewDeductFeeDecorator(ak, supplyKeeper),
+		cosmosante.NewSigGasConsumeDecorator(ak, sigGasConsumer),
+		cosmosante.NewSigVerificationDecorator(ak),
+		cosmosante.NewIncrementSequenceDecorator(ak),
+	)
+}
+
+type MinFeeDecorator struct {
+	feesKeeper fees.Keeper
+}
+
+func NewMinFeeDecorator(feesKeeper fees.Keeper) MinFeeDecorator {
+	return MinFeeDecorator{
+		feesKeeper: feesKeeper,
+	}
+}
+
+func (mfd MinFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	// all transactions must be of type auth.StdTx
+	stdTx, ok := tx.(types.StdTx)
+	if !ok {
+		// Set a gas meter with limit 0 as to prevent an infinite gas meter attack
+		// during runTx.
+		newCtx = setGasMeter(simulate, ctx, 0)
+		return newCtx, errors.New("tx must be StdTx")
```

Comment:
Is this something we need to do here, or is this a check that might have already happened previously inside the ante chain?

Reply:
Yup, others ante checks for particular txs types, so I think we needed that to be sure we're having only stdTx